frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/EditFieldDrawer/edit-fieldtype/EditTable/EditTable.tsx

```diff
@@ -0,0 +1,200 @@
+import { useCallback, useMemo } from 'react'
+import {
+  Controller,
+  DeepPartial,
+  FormProvider,
+  UnpackNestedValue,
+  useFormState,
+} from 'react-hook-form'
+import { FormControl, Stack, StackDivider } from '@chakra-ui/react'
+import { extend, pick } from 'lodash'
+
+import { Column, ColumnDto, TableFieldBase } from '~shared/types/field'
+
+import { REQUIRED_ERROR } from '~constants/validation'
+import { createBaseValidationRules } from '~utils/fieldValidation'
+import FormErrorMessage from '~components/FormControl/FormErrorMessage'
+import FormLabel from '~components/FormControl/FormLabel'
+import Input from '~components/Input'
+import NumberInput from '~components/NumberInput'
+import Textarea from '~components/Textarea'
+import Toggle from '~components/Toggle'
+
+import { isTemporaryColumnId } from '~features/admin-form/create/builder-and-design/utils/columnCreation'
+
+import { DrawerContentContainer } from '../common/DrawerContentContainer'
+import { FormFieldDrawerActions } from '../common/FormFieldDrawerActions'
+import { EditFieldProps } from '../common/types'
+import { useEditFieldForm } from '../common/useEditFieldForm'
+
+import { EditTableColumns } from './EditTableColumns'
+
+const EDIT_TABLE_FIELD_KEYS = [
+  'title',
+  'description',
+  'required',
+  'addMoreRows',
+  'minimumRows',
+  'maximumRows',
+  'columns',
+] as const
+
+export type EditTableInputs = Omit<
+  Pick<TableFieldBase, typeof EDIT_TABLE_FIELD_KEYS[number]>,
+  'columns' | 'maximumRows' | 'minimumRows'
+> & {
+  // Every column must have an ID for react-table to render.
+  columns: ColumnDto[]
+  maximumRows: string | number
+  minimumRows: string | number
+}
+
+export type EditTableProps = EditFieldProps<TableFieldBase>
+
+const transformTableFieldToEditForm = (
+  field: TableFieldBase,
+): UnpackNestedValue<DeepPartial<EditTableInputs>> => {
+  const nextMaxRows = field.maximumRows || ''
+  const nextMinRows = field.minimumRows || ''
+
+  return {
+    ...pick(field, EDIT_TABLE_FIELD_KEYS),
+    maximumRows: nextMaxRows,
+    minimumRows: nextMinRows,
+  }
+}
+
+const transformTableEditFormToField = (
+  inputs: EditTableInputs,
+  originalField: TableFieldBase,
+): TableFieldBase => {
+  return extend({}, originalField, inputs)
+}
+
+export const EditTable = ({ field }: EditTableProps): JSX.Element => {
+  const preSubmitTransform = useCallback(
+    ({ columns, ...rest }: EditTableInputs, output: TableFieldBase) => {
+      const columnsWithoutTempIds: Column[] = columns.map((column) => {
+        const { _id, ...restColumn } = column
+        if (isTemporaryColumnId(_id)) {
+          return restColumn
```

Comment:
would be good to add some comments here to explain - this is so that the temporary IDs don't get saved to the db right?